torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -199,7 +199,9 @@ def runtime_wrapper(args: List[Any]):
             for idx in indices_of_inps_to_detach:
                 if isinstance(args_[idx], torch.Tensor):
                     args_[idx] = args_[idx].detach()
-            with torch.autograd._force_original_view_tracking(True):
+            with torch.autograd._force_original_view_tracking(
+                True
+            ), torch.enable_grad():
```

Comment:
Can you add a comment explaining why we need the `enable_grad` here (since it's pretty subtle - it's possible for a user to run a region of code under `no_grad()`, but still require us to compute gradients for it)

Reply:
QQ: why doesn't the "enable_grad" in the subregion of the function apply `enable_grad`. I.e. why when calling compiled_fn, do we not automatically enable grad as needed? Why do we need to explicitly set it here? 

(This is assuming we still apply the other change in this PR, so that trace_joint is True when any elements require grad).

Reply:
Yeah that is a good question. At runtime, we are effectively just executing a giant `torch.autograd.Function` containing our compiled fw and bw graphs (this is how we get training support).

But normally, the entire point of `no_grad` is to say "actually I do not want to compute any gradients". So without this change, if a user ran this code:
```
with torch.no_grad():
    return f(x)
```

That would effectively turn into this:
```
with torch.no_grad():
    return CompiledFunction.apply(x)
```

If `torch.is_grad_enabled() == False` when we run an `autograd.Function`, the `autograd.Function` will  respect the global grad state and all of its outputs will be set to not require grad. However, we know that we actually do need to compute some gradients here (even if the global grad state happens to be False).

Let me know if that is still unclear!